modules/local/locidex_select.nf

```diff
@@ -0,0 +1,276 @@
+/*
+Locidex provides the option to select a database from a group of databases, by
+using a manifest file. To prevent copying all databases each time an allele database
+needs to be selected this file will read only the "manifest" file of the databases and
+to pick the correct allele scheme.
+
+
+Locidex Manifest is setup as:
+    {
+        db_name: [
+            {
+                "path": path/to/db/relative/to/manifest
+                # DB Config data, the newest db data will be selected as versions are not standardized
+                "config": {
+                    "db_name": "Locidex Database 1",
+                    "db_version": "1.0.0",
+                    "db_date": "yyyy-MM-dd",
+                    "db_author": "test1",
+                    "db_desc": "test1",
+                    "db_num_seqs": 53,
+                    "is_nucl": true,
+                    "is_prot": true,
+                    "nucleotide_db_name": "nucleotide",
+                    "protein_db_name": "protein"
+                }
+            }
+        ]
+    }
+*/
+
+import groovy.json.JsonSlurper
+import groovy.json.JsonBuilder
+import java.text.SimpleDateFormat
+
+
+process LOCIDEX_SELECT {
+    tag "$meta.id"
+    label "process_single"
+
+    input:
+    tuple val(meta), val(top_hit), val(contigs)
+    val manifest // This is a json file to be parsed
+    val config // optional path to a database file
+
+    output:
+    tuple val(meta), val(contigs), val(scheme), val(paired_p), emit: db_data
+    tuple val(meta), path(output_config), emit: config_data
+
+    exec:
+
+    report_name = "${meta.id}_${params.locidex.db_config_output_name}"
+    output_config = task.workDir.resolve(report_name)
+
+    if(params.allele_scheme == null && params.locidex.allele_database == null){
+        error("Allele calling is enabled but there is no allele scheme or locidex allele database location present.")
+    }
+
+    if(config){ // non null channel should evaluate to true
+        if(manifest){
+            log.warn "A database manifest file was passed along with a config file. Using the allele database specified by '--allele_scheme' "
+        }
+        // Create output summary data for a locidex allele database if an allele scheme is passed in directly
+        def jsonSlurper = new JsonSlurper()
+        String json_data = config.text
```

Comment:
in this case it seems `val` does not actually coerce the file staged in the workflow to any string value. So it still exists as a path object when passed to the pipeline.

However I this may be the source of some issues in blob storage.

Reply:
Just tested this change out, if I revert to using `file(config)` e.g. calling it again an error is thrown. 

passing in config as a value seems to pass in a path that can be resolved in blob storage (some how)

If other issues persist, I may parse the file within the workflow code and pass it in as a string value if any errors show up.

Reply:
Thanks so much for testing this out. You are right that using `val` coerces the value to a path object. See https://github.com/nextflow-io/nextflow/issues/1678#issuecomment-927913406